Sources/XcodeCompilationDatabaseCore/XcodeCompilationDatabase.swift

```diff
@@ -0,0 +1,209 @@
+import Foundation
+
+public func shlex(_ input: String) -> [String] {
+    var result: [String] = []
+    
+    var accumulator = ""
+    let unitQuote = ""
+    var quote = unitQuote
+    var escape = true
+    
+    func evictAccumulator() {
+        if accumulator != "" {
+            result.append(accumulator)
+            accumulator = "";
+        }
+    }
+
+    func isspace(_ c: String) -> Bool {
+        return c == " "
+    }
+
+    let doubleQuote = "\""
+    let singleQuote = "\'"
+
+    for c in input {
+        let token = String(c)
+        if escape {
+            escape = false;
+            accumulator += token
+        } else if token == "\\" {
+            escape = true;
+        } else if (quote == unitQuote && token == singleQuote) ||
+                   (quote == unitQuote && token == unitQuote) {
+            quote = token
+        } else if (quote == singleQuote && token == singleQuote) ||
+                   (quote == doubleQuote  && token == doubleQuote) {
+            quote = unitQuote
+        } else if !isspace(token) || quote != unitQuote {
+            accumulator += String(token)
+        } else {
+            evictAccumulator()
+        }
+    }
+    evictAccumulator()
+    return result;
+}
+
+/// Check if the first non " " argument is clang
+func isCC(lexed: [String]) -> Bool {
```

Comment:
None of these naive checks will work with custom compilers ( for clang and swiftc ), that is an edge case that this program doesn't need to handle ATM. It should be at least called out.